packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -43,21 +41,38 @@ export function isNodeCompat(
 }
 
 /**
- * If the current environment needs Node.js compatibility,
- * then inject the necessary global polyfills into the code.
+ * Get a set of module specifiers for all possible Node.js compat polyfill entry-points
+ */
+export function getNodeCompatEntries() {
+	const entries = new Set<string>(Object.values(env.alias));
+	for (const globInject of Object.values(env.inject)) {
+		if (typeof globInject === "string") {
+			entries.add(globInject);
+		} else {
+			entries.add(globInject[0] as string);
+		}
+	}
+	for (const external of env.external) {
+		entries.delete(external);
+	}
+	return entries;
+}
+
+/**
+ * Get the necessary global polyfills to inject into the entry-point of the user's code.
  */
 export function injectGlobalCode(id: string, code: string) {
 	const injectedCode = Object.entries(env.inject)
 		.map(([globalName, globalInject]) => {
 			if (typeof globalInject === "string") {
 				const moduleSpecifier = globalInject;
 				// the mapping is a simple string, indicating a default export, so the string is just the module specifier.
-				return `import var_${globalName} from "${CLOUDFLARE_VIRTUAL_PREFIX}${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName};\n`;
+				return `import var_${globalName} from "${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName};\n`;
 			}
 
 			// the mapping is a 2 item tuple, indicating a named export, made up of a module specifier and an export name.
 			const [moduleSpecifier, exportName] = globalInject;
-			return `import var_${globalName} from "${CLOUDFLARE_VIRTUAL_PREFIX}${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName}.${exportName};\n`;
+			return `import var_${globalName} from "${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName}.${exportName};\n`;
```

Comment:
```suggestion
			assert(moduleSpecifier && exportName, `Unexpected error: globalInject is not a string or tuple`);
			return `import var_${globalName} from "${moduleSpecifier}";\nglobalThis.${globalName} = var_${globalName}.${exportName};\n`;
```

Reply:
4de9f48d9c6af8eb2319b4cb8ce5f4f892b99c53

Reply:
Is this necessary? It'll totally break anyway if these are not defined.

Reply:
Up to you. I would rather control the error that's reported if it breaks. Though, as I also mentioned it would be better if unenv typed tuples as `[string, string]`.

Reply:
If the types were correct then we wouldn't add the assertion, and we'd get the same error as right now.
But given that you asked, I'll add it 🤗

Reply:
9471f87860fef681269c0b8d7a31335c1ebd952f